h2/src/test/org/h2/test/unit/TestPgServer.java

```diff
@@ -695,4 +699,34 @@ private void testOtherPgClients() throws SQLException {
             conn0.close();
         }
     }
+
+    private void testArray() throws Exception {
+        if (!getPgJdbcDriver()) {
+            return;
+        }
+
+        Server server = createPgServer(
+                "-ifNotExists", "-pgPort", "5535", "-pgDaemon", "-key", "pgserver", "mem:pgserver");
+        try (
+                Connection conn = DriverManager.getConnection(
+                        "jdbc:postgresql://localhost:5535/pgserver", "sa", "sa");
+                Statement stat = conn.createStatement();
+        ) {
+            stat.execute("CREATE TABLE test (id int primary key, x1 int array, x2 varchar array)");
+            stat.execute("INSERT INTO test (id, x1, x2) VALUES (1, ARRAY[2, 3], ARRAY['4', '5'])");
+            try (ResultSet rs = stat.executeQuery(
+                    "SELECT x1, x2 FROM test WHERE id = 1")) {
+                assertTrue(rs.next());
+                Object[] arr = (Object[]) rs.getArray(1).getArray();
+                assertEquals("2", arr[0]);
+                assertEquals("3", arr[1]);
+                arr = (Object[]) rs.getArray(2).getArray();
+                assertEquals("4", arr[0]);
+                assertEquals("5", arr[1]);
+            }
```

Comment:
Please, add tests for special characters and for array assignments via parameters of `PreparedStatement` and all necessary code for them.

Reply:
Prepare statement for an Array does not work like

    PreparedStatement ps = conn.prepareStatement("INSERT INTO test (id, x1) VALUES (1, ?)")
    Array arr = conn.createArrayOf("VARCHAR", new Object[] {"'a'b'c'", "d\"e", "{,}"});

pg-jdbc seems to generate an SQL that H2 does not support:

    SELECT t.oid, t.typname   FROM pg_catalog.pg_type t  JOIN pg_catalog.pg_namespace n ON t.typnamespace = n.oid WHERE t.typelem = (SELECT oid FROM pg_catalog.pg_type WHERE typname = $1) AND substring(t.typname, 1, 1) = '_' AND t.typlen = -1 AND (n.nspname = $2 OR $3 AND n.nspname = ANY (current_schemas(true))) ORDER BY t.typelem DESC LIMIT 1

Reply:
Does it work with `setObject(column, someArray)`?

Reply:
`setObject(column, String[], Types.ARRAY)` works, and send:

    INSERT INTO test (id, x1) VALUES (1, $1)
    $1 = {"'a'b'c'", "d\"e", "{,}"}

But H2 treated as an array with one element `{"{\"'a'b'c'\", \"d\\\"e\", \"{,}\"}"}`.

Reply:
Hypothetically `PgServerThread.setParameter()` can parse `PG_TYPE_TEXTARRAY` parameters in that way.